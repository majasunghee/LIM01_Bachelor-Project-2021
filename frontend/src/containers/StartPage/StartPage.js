import React, { useState } from 'react';
import { Link, Redirect } from 'react-router-dom';
import { Toolbar, Typography } from '@material-ui/core';
import Appbar from '@material-ui/core/AppBar';
import pickleLogo from '../../assets/images/pickleLogo.png';
import ellipse from '../../assets/images/ellipse.png';
import tree from '../../assets/images/tree.png';
import forest from '../../assets/images/forest.png';
import useStyles from './styles';
import SearchBar from '../../components/SearchBar/SearchBar';

const StartPage = () => {
  const classes = useStyles();
  // eslint-disable-next-line no-unused-vars
  const [id, setId] = useState(null);
  const [redirect, setRedirect] = useState(false);
  const [error, setError] = useState(false);

  const onChange = (e) => setId(e.target.value);

  function playSet() {
    if (!id) {
      setError(true);
    } else {
      setRedirect(true);
    }
  }

  return (
    <div className={classes.root}>
      <Appbar position="relative" className={classes.navbar} elevation={0}>
        <Toolbar>
          <Typography variant="h6" className={classes.right}>
            <Link to="/login" className={classes.title}>
              Log inn
            </Link>
            <Link to="/signup" className={classes.title}>
              Registrer deg
            </Link>
          </Typography>
        </Toolbar>
      </Appbar>
      <img src={ellipse} alt="ellipse" className={classes.ellipse} />
      <div className={classes.logoBox}>
        <img src={pickleLogo} alt="pickle logo" className={classes.logoImg} />
        <Typography variant="h3" className={classes.logoText}>
          DiPICKLE
        </Typography>
      </div>
      <SearchBar onChange={onChange} playSet={playSet} error={error} />
      <img src={forest} alt="forest" className={classes.forest} />
      <img src={tree} alt="tree" className={classes.tree} />
      {redirect ? (
        <Redirect
          to={{
            pathname: '/sets',
            state: { id },
          }}
        />
      ) : (
        <> </>
      )}
    </div>
  );
};

export default StartPage;
