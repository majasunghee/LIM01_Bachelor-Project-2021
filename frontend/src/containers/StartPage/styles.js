import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles((theme) => ({
  root: {
    textAlign: 'center',
    height: '100vh',
  },
  navbar: {
    backgroundColor: 'transparent',
    elevation: 'none',
  },
  searchBar: {
    margin: 'auto',
    maxWidth: '40vw',
    [theme.breakpoints.down('xs')]: {
      maxWidth: '60vw',
    },
  },
  title: {
    flexGrow: 1,
    color: '#367051',
    fontWeight: '600',
    float: 'right',
    padding: theme.spacing(1),
    textDecoration: 'none',
    '&:hover': {
      backgroundColor: '#47956A',
      color: 'white',
      borderRadius: '15px',
    },
  },
  right: {
    flex: 1,
  },
  logoBox: {
    lineHeight: '200px',
    justifyContent: 'center',
    [theme.breakpoints.down('xs')]: {
      marginTop: '15vh',
      marginLeft: theme.spacing(2),
      marginRight: theme.spacing(2),
    },
  },
  logoImg: {
    verticalAlign: 'middle',
    [theme.breakpoints.down('xs')]: {
      maxWidth: '70%',
    },
  },
  logoText: {
    display: 'inline-block',
    border: '3px solid #0F6D5F',
    boxSizing: 'border-box',
    fontFamily: 'Roboto',
    fontWeight: '200',
    color: '#0F6D5F',
    paddingLeft: theme.spacing(2),
    paddingRight: theme.spacing(2),
    [theme.breakpoints.down('xs')]: {
      maxWidth: '70%',
    },
  },
  ellipse: {
    display: 'block',
    position: 'absolute',
    top: '0',
    [theme.breakpoints.up('xs')]: {
      maxWidth: '20vw',
    },
    [theme.breakpoints.down('xs')]: {
      top: '0',
      maxWidth: '40vw',
    },
  },
  tree: {
    display: 'block',
    position: 'absolute',
    bottom: '0',
    right: '10vw',
    [theme.breakpoints.up('xs')]: {
      maxWidth: '16vw',
    },
    [theme.breakpoints.down('xs')]: {
      right: '5vw',
      maxWidth: '30vw',
    },
  },
  forest: {
    display: 'block',
    position: 'absolute',
    bottom: '0',
    left: '12vw',
    [theme.breakpoints.up('xs')]: {
      maxWidth: '28vw',
    },
    [theme.breakpoints.down('xs')]: {
      left: '5vw',
      maxWidth: '60vw',
    },
  },
}));
export default useStyles;
