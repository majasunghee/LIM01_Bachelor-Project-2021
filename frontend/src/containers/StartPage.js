import React from 'react';
import { Link } from 'react-router-dom';
import { makeStyles } from '@material-ui/core/styles';
import {
  Button,
  Grid,
  TextField,
  Toolbar,
  Typography,
} from '@material-ui/core';
import Appbar from '@material-ui/core/AppBar';
import pickleLogo from '../assets/images/pickleLogo.png';
import ellipse from '../assets/images/ellipse.png';
import tree from '../assets/images/tree.png';
import forest from '../assets/images/forest.png';

const useStyles = makeStyles((theme) => ({
  root: {
    textAlign: 'center',
    height: '100vh',
    background: 'linear-gradient(90deg, #53A77A 1.46%, #80D197 100%)',
  },
  navbar: {
    backgroundColor: 'transparent',
    elevation: 'none',
  },
  title: {
    flexGrow: 1,
    color: '#367051',
    fontWeight: '600',
    float: 'right',
    margin: theme.spacing(1),
    textDecoration: 'none',
  },
  right: {
    flex: 1,
  },
  logoBox: {
    lineHeight: '200px',
    justifyContent: 'center',
  },
  logoImg: {
    verticalAlign: 'middle',
  },
  logoText: {
    display: 'inline-block',
    border: '3px solid #0F6D5F',
    boxSizing: 'border-box',
    fontFamily: 'Roboto',
    fontWeight: '200',
    color: '#0F6D5F',
    paddingLeft: theme.spacing(2),
    paddingRight: theme.spacing(2),
  },
  searchBox: {
    margin: 'auto',
    width: '40vw',
  },
  search: {
    backgroundColor: 'white',
    border: '2px solid #0F6D5F',
    boxSizing: 'border-box',
    borderRadius: '5px',
    margin: theme.spacing(4),
  },
  btn: {
    backgroundColor: '#F7B733',
    color: 'black',
    fontWeight: 'bold',
    filter: 'drop-shadow(0px 4px 10px rgba(0, 0, 0, 0.15))',
  },
  ellipse: {
    width: '20vw',
    display: 'block',
    position: 'absolute',
    top: '0',
  },
  tree: {
    width: '16vw',
    display: 'block',
    position: 'absolute',
    bottom: '0',
    right: '10vw',
  },
  forest: {
    width: '28vw',
    display: 'block',
    position: 'absolute',
    bottom: '0',
    left: '12vw',
  },
}));

const StartPage = () => {
  const classes = useStyles();
  return (
    <div className={classes.root}>
      <Appbar position="relative" className={classes.navbar} elevation={0}>
        <Toolbar>
          <Typography variant="h6" className={classes.right}>
            <Link to="/login" className={classes.title}>
              Log inn
            </Link>
            <Link to="/signup" className={classes.title}>
              Registrer deg
            </Link>
          </Typography>
        </Toolbar>
      </Appbar>
      <img src={ellipse} alt="ellipse" className={classes.ellipse} />
      <div className={classes.logoBox}>
        <img src={pickleLogo} alt="pickle logo" className={classes.logoImg} />
        <Typography variant="h3" className={classes.logoText}>
          DiPLICKLE
        </Typography>
      </div>
      <div className={classes.searchBox}>
        <Grid
          container
          spacing={0}
          direction="row"
          alignItems="center"
          justify="center"
        >
          <Grid item xs={9}>
            <TextField
              className={classes.search}
              type="text"
              placeholder="Finn oppgavesett"
              variant="outlined"
              margin="dense"
              fullWidth
            />
          </Grid>
          <Grid item xs={1} />
          <Grid item xs={2}>
            <Button fullWidth variant="contained" className={classes.btn}>
              SÃ¸k
            </Button>
          </Grid>
        </Grid>
      </div>
      <img src={forest} alt="forest" className={classes.forest} />
      <img src={tree} alt="tree" className={classes.tree} />
    </div>
  );
};

export default StartPage;
