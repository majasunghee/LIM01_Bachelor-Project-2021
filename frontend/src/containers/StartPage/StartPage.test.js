/* eslint-disable no-undef */
import React from 'react';
import { render, screen, fireEvent, act } from '@testing-library/react';
import { BrowserRouter as Router, MemoryRouter } from 'react-router-dom';
import StartPage from './StartPage';

describe('The StartPage component', () => {
  it('should render', () => {
    // arrange
    const { asFragment } = render(
      <Router>
        <StartPage />
      </Router>
    );

    // act
    const html = asFragment();

    // assert
    expect(html).toMatchSnapshot();
  });

  it('should contain the title text "DiPICKLE"', () => {
    const { getByText } = render(
      <Router>
        <StartPage />
      </Router>
    );

    const title = getByText('DiPICKLE');

    expect(title.tagName.toLowerCase()).toEqual('h3');
  });
});

describe('The Link component should', () => {
  it('contain a navigation link to the login page', () => {
    const { getByText } = render(
      <MemoryRouter>
        <StartPage />
      </MemoryRouter>
    );

    const link = getByText('Logg inn');

    expect(link).toBeInTheDocument();
    expect(link).toHaveAttribute('href', '/login');
  });
  it('contain a navigation link to the Signup page', () => {
    const { getByText } = render(
      <MemoryRouter>
        <StartPage />
      </MemoryRouter>
    );

    const link = getByText('Registrer deg');

    expect(link).toBeInTheDocument();
    expect(link).toHaveAttribute('href', '/signup');
  });
});

describe('Searchbar', () => {
  test('show an error message on illegal input', async () => {
    await act(async () =>
      render(
        <Router>
          <StartPage />
        </Router>
      )
    );

    const inputEl = screen.getByPlaceholderText('Tast inn settets ID');
    fireEvent.change(inputEl, { target: { value: 'hei' } });
    const button2 = screen.getByText('SPILL');
    fireEvent.click(button2);

    await screen.findByText('Settet finnes ikke.');
    const errorMessage = screen.getByText('Settet finnes ikke.');

    expect(errorMessage.tagName.toLowerCase()).toEqual('span');
  });
});
