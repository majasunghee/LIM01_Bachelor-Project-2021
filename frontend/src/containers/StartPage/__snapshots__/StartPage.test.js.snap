// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The StartPage component should render 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-root-1"
  >
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionRelative MuiAppBar-colorPrimary makeStyles-navbar-2 MuiPaper-elevation0"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <h6
          class="MuiTypography-root makeStyles-right-5 MuiTypography-h6"
        >
          <a
            class="makeStyles-title-4"
            href="/login"
          >
            Logg inn
          </a>
          <a
            class="makeStyles-title-4"
            href="/signup"
          >
            Registrer deg
          </a>
        </h6>
      </div>
    </header>
    <img
      alt="ellipse"
      class="makeStyles-ellipse-9"
      src="ellipse.png"
    />
    <div
      class="makeStyles-logoBox-6"
    >
      <img
        alt="pickle logo"
        class="makeStyles-logoImg-7"
        src="pickleLogo.png"
      />
      <h3
        class="MuiTypography-root makeStyles-logoText-8 MuiTypography-h3"
      >
        DiPICKLE
      </h3>
    </div>
    <div
      class="makeStyles-searchBar-3"
    >
      <div
        class="makeStyles-searchBox-12"
      >
        <form>
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10"
            >
              <div
                class="MuiFormControl-root MuiTextField-root makeStyles-search-13 MuiFormControl-marginDense MuiFormControl-fullWidth"
              >
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                    placeholder="Tast inn settets ID"
                    required=""
                    type="text"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="PrivateNotchedOutline-root-17 MuiOutlinedInput-notchedOutline"
                    style="padding-left: 8px;"
                  >
                    <legend
                      class="PrivateNotchedOutline-legend-18"
                      style="width: 0.01px;"
                    >
                      <span>
                        â€‹
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-btn-14 MuiButton-containedPrimary MuiButton-fullWidth"
                tabindex="0"
                type="submit"
              >
                <span
                  class="MuiButton-label"
                >
                  SPILL
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
    <img
      alt="forest"
      class="makeStyles-forest-11"
      src="forest.png"
    />
    <img
      alt="tree"
      class="makeStyles-tree-10"
      src="tree.png"
    />
  </div>
</DocumentFragment>
`;
